(Near)Semiring & Lattice morphisms

conal opened the pull request:
Part of issue #1297. Work in progress. Feedback most appreciated (as picky as you want!).

MatthewDaggitt commented on src/Algebra/Bundles.agda:
```diff
@@ -341,6 +341,30 @@ record DistributiveLattice c ℓ : Set (suc (c ⊔ ℓ)) where
+    _*_        : Op₂ Carrier
+    0#         : Carrier
+
+  +-rawMonoid : RawMonoid c ℓ
```
I would probably open this publicly and export `+-rawMagma` as well?

conal replied:
Good catch. Fixed. Thanks.

MatthewDaggitt commented on src/Algebra/Morphism/Structures.agda:
```diff
@@ -177,6 +177,90 @@ module GroupMorphisms (G₁ : RawGroup a ℓ₁) (G₂ : RawGroup b ℓ₂) wher
+      ; injective            = injective
+      }
+
+    -- TODO: Why isRelMonomorphism specifically here?
```
Because it's the only thing not exported by `open IsNearSemiringHomomorphism isNearSemiringHomomorphism public` above

MatthewDaggitt reviewed (commented):
Looks good! After making `+-rawMagma` change, we can either merge this in immediately or you can add the semiring ones to this PR. If you'd prefer the former, then you'll need to add an entry to the CHANGELOG.

conal commented:
@MatthewDaggitt Thanks for the guidance! I addressed the issues you mentioned in review comments, plus a few cosmetics. Added Semiring morphisms and CHANGELOG entry. I think this PR is ready to merge if you (and others?) see fit.

conal commented:
Since my head is in this code, I made changes for `Lattice` as well and updated the CHANGELOG and PR name accordingly. Please let me know how this latest commit looks to you, and add review comments or merge if ready.

BTW, in `Algebra.Bundles`, I noticed that `Lattice` has a `setoid` operation, which seems useful. Do you know why other records defined in that module don’t have `setoid` (say, via `Magma`)?

MatthewDaggitt commented:
> Since my head is in this code, I made changes for Lattice as well and updated the CHANGELOG and PR name accordingly. Please let me know how this latest commit looks to you, and add review comments or merge if ready.

Looks good to me! Thanks for the great PR! We usually leave the PR up for a few days in case anyone else has some comments. Assuming no one does, I'll merge it in early next week :+1: 

> BTW, in Algebra.Bundles, I noticed that Lattice has a setoid operation, which seems useful. Do you know why other records defined in that module don’t have setoid (say, via Magma)?

I don't know. Seems probably like an omission. Feel free to open another PR adding it in if you would find it useful!

conal commented:
> BTW, in `Algebra.Bundles`, I noticed that `Lattice` has a `setoid` operation, which seems useful. Do you know why other records defined in that module don’t have `setoid` (say, via `Magma`)?

I found `setoid` in `IsMagma`.
